cunumeric/array.py

```diff
@@ -172,6 +176,68 @@ def maybe_convert_to_np_ndarray(obj: Any) -> Any:
     return obj
 
 
+def check_writeable(arr: Union[ndarray, tuple[ndarray, ...], None]) -> None:
+    """
+    Check if the current array is writeable
+    This check needs to be manually inserted
+    with consideration on the behavior of the corresponding method
+    """
+    if arr is None:
+        return
+    check_list = (arr,) if not isinstance(arr, tuple) else arr
+    if any(not arr.flags.writeable for arr in check_list):
+        raise ValueError("array is not writeable")
+
+
+class flagsobj:
+    def __init__(self, array: Any) -> None:
+        object.__setattr__(self, "_array", array)
```

Comment:
Does the simple form (`self._array = array`) not work here?

Reply:
not with the current `__setattr__`
```
  cunumeric/array.py:194: in __init__
       self._array = array
   cunumeric/array.py:217: in __setattr__
       flags = self._array.__array__().flags
   cunumeric/array.py:209: in __getattr__
       flags = self._array.__array__().flags
   cunumeric/array.py:209: in __getattr__
       flags = self._array.__array__().flags
   E   RecursionError: maximum recursion depth exceeded in comparison
   !!! Recursion detected (same locals & position)
```

Reply:
https://github.com/nv-legate/cunumeric/pull/759/commits/4fb6e27dedbf310bfc16ce3f0da6d0520300df50 (added a comment)